### PR TITLE
support double tap

### PR DESCRIPTION
**Pull request by Laserlicht**
Support double "click" on touch.

### CODE REVIEW

**Comment by Laserlicht** [comment]
tested; works

File: client/eventsSDL/InputSourceTouch.cpp [hunk 1]
```diff
@@ -181,8 +181,15 @@ void InputSourceTouch::handleEventFingerUp(const SDL_TouchFingerEvent & tfinger)
 		case TouchState::TAP_DOWN_SHORT:
 		{
 			GH.input().setCursorPosition(convertTouchToMouse(tfinger));
-			GH.events().dispatchMouseLeftButtonPressed(convertTouchToMouse(tfinger), params.touchToleranceDistance);
-			GH.events().dispatchMouseLeftButtonReleased(convertTouchToMouse(tfinger), params.touchToleranceDistance);
+			if(tfinger.timestamp - lastLeftClickTimeTicks < params.doubleTouchTimeMilliseconds && (convertTouchToMouse(tfinger) - lastLeftClickPosition).length() < params.doubleTouchToleranceDistance)
+				GH.events().dispatchMouseDoubleClick(convertTouchToMouse(tfinger));
```
Comment (IvanSavenko) [inline client/eventsSDL/InputSourceTouch.cpp:185]: We may need to add `dispatchMouseLeftButtonReleased` here.
Check double-clicking on any button that can be clicked multiple times, for example "next hero" button.

Also - add touch tolerance to double-click?

```suggestion
				GH.events().dispatchMouseDoubleClick(convertTouchToMouse(tfinger));
				GH.events().dispatchMouseLeftButtonReleased(convertTouchToMouse(tfinger), params.touchToleranceDistance);
```

**Comment by Nordsoft91** [comment]
What about supporting triple touch? Can be used to emulate "space" on touchscreen

**Review by IvanSavenko — APPROVED** [review]

**Comment by Laserlicht** [comment]
> What about supporting triple touch? Can be used to emulate "space" on touchscreen

Too much and unintuitive I think.

**Comment by IvanSavenko** [comment]
> What about supporting triple touch? Can be used to emulate "space" on touchscreen

Currently I want to add radial menu similar to one we have for creature management to adventure map.
 
Object revisit is not the only missing action on adventure map, so sounds like a good idea. Wanted to do this for 1.3, but haven't find time for it.

**Pull request merged**